src/lib/components/eventModal.svelte

```diff
@@ -1,184 +1,241 @@
+        ...services.flatMap((s) =>
+            s.resources.flatMap((r) => r.actions.flatMap((a) => a.attributes ?? []))
+        )
+    ];
 
-    let selectedService: Service = null;
-    let selectedResource: Resource = null;
-    let selectedAction: Action = null;
-    let selectedAttribute: string = null;
-    let helper: string = null;
-    let inputData: string = null;
-    let showInput = false;
+    // Helpers
+    const isService = (s: string) => serviceNames.includes(s);
+    const isResource = (s: string) => resourceNames.includes(s);
+    const isAction = (s: string) => actionNames.includes(s);
+    const isAttribute = (s: string) => attributeNames.includes(s);
 
     const dispatch = createEventDispatcher();
 
     function create() {
         show = false;
-        dispatch('created', copyValue);
+        dispatch('created', inputValue);
     }
 
-    const actions = [{ name: 'create' }, { name: 'update' }, { name: 'delete' }];
-
-    const services = [
-        {
-            name: 'buckets',
-            resources: [{ name: 'files', actions }],
-            actions
-        },
-        {
-            name: 'databases',
-            resources: [
-                { name: 'collections', actions },
-                { name: 'documents', actions }
-            ],
-            actions
-        },
-        {
-            name: 'functions',
-            resources: [
-                { name: 'deployments', actions },
-                { name: 'executions', actions }
-            ],
-            actions
-        },
-        {
-            name: 'teams',
-            resources: [{ name: 'memberships', actions }],
-            actions: [
-                { name: 'create' },
-                { name: 'update', attributes: ['status'] },
-                { name: 'delete' }
-            ]
-        },
-        {
-            name: 'users',
-            resources: [
-                { name: 'recovery', actions: [{ name: 'create' }, { name: 'delete' }] },
-                { name: 'sessions', actions: [{ name: 'create' }, { name: 'delete' }] },
-                { name: 'verifications', actions: [{ name: 'create' }, { name: 'delete' }] }
-            ],
-            actions: [
-                { name: 'create' },
-                { name: 'update', attributes: ['email', 'name', 'password', 'status', 'prefs'] },
-                { name: 'delete' }
-            ]
+    function select(field: 'service', value: EventService);
+    function select(field: 'resource', value: EventResource);
+    function select(field: 'action', value: EventAction);
+    function select(field: 'attribute', value: string);
+    function select(field: string, value: string | EventService | EventResource | EventAction) {
+        if (typeof value === 'string') {
+            selected[field as 'attribute'] = selected[field] === value ? null : value;
+        } else {
+            selected[field as any] = (selected[field] as any)?.name === value.name ? null : value;
```

Comment:
we should probably use `unknown` over `any`

Reply:
Removed any's